tests/gccgo/task.yaml

```diff
@@ -0,0 +1,14 @@
+summary: Check that snapd builds with gccgo
+prepare: |
+    echo Installing gccgo-6 and pretending it is the default go
+    sudo apt install -y gccgo-6
+    sudo ln -s /usr/bin/go-6 /usr/local/bin/go
+restore: |
+    rm -f /usr/local/bin/go
+    sudo apt-get autoremove -y gccgo-6
+execute: |
+    echo Ensure we really build with gccgo
+    go version|grep gccgo
+    echo Build the deb with gccgo and run the tests as part of the build
+    sudo -i -u test /bin/sh -c "cd /gopath/src/github.com/snapcore/snapd && dpkg-buildpackage -Zgzip"
```

Comment:
Sorry, just one small request here. The tests were running out of disk space, so I just increased the disk size which should hopefully fix it, but can we please also have the `-tc` flag in such builds, so it clears up after building (is that the right flag?)? Also, this will leave packages behind, right?  If so, we should also remove these on restore as we won't be using them after the test.

I've also added the `-tc` flag on the main package build in the project prepare, on #1347.